pytorch_lightning/loggers/neptune.py

```diff
@@ -114,160 +140,229 @@ def any_lightning_module_function_or_hook(self):
+                )
+                self._run_instance['source_code/integrations/neptune-pytorch-lightning'] = __version__
+            except NeptuneLegacyProjectException as e:
+                raise TypeError(f"""
+                    Project {self._project} has not been imported to new structure yet.
+                    You can still integrate it with `NeptuneLegacyLogger`.
+                    """) from e
+
+        return self._run_instance
 
     @rank_zero_only
     def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
+        r"""
+        Log hyper-parameters to the run.
+
+        Params will be logged using the ``param__`` scheme, for example: ``param__batch_size``, ``param__lr``.
+
+        **Note**
+
+        You can also log parameters by directly using the logger instance:
+        ``neptune_logger.experiment['model/hyper-parameters'] = params_dict``.
+
+        In this way you can keep hierarchical structure of the parameters.
+
+        Args:
+            params: `dict`.
+                Python dictionary structure with parameters.
+
+        Example::
+
+            from pytorch_lightning.loggers import NeptuneLogger
+
+            PARAMS = {
+                'batch_size': 64,
+                'lr': 0.07,
+                'decay_factor': 0.97
+            }
+
+            neptune_logger = NeptuneLogger(
+                api_key='ANONYMOUS',
+                close_after_fit=False,
+                project='common/new-pytorch-lightning-integration'
+            )
+
+            neptune_logger.log_hyperparams(PARAMS)
+        """
         params = self._convert_params(params)
         params = self._flatten_dict(params)
         for key, val in params.items():
-            self.experiment.set_property(f'param__{key}', val)
+            self.run[f'param__{key}'] = val
 
     @rank_zero_only
     def log_metrics(self, metrics: Dict[str, Union[torch.Tensor, float]], step: Optional[int] = None) -> None:
         """
-        Log metrics (numeric values) in Neptune experiments.
+        Log metrics (numeric values) in Neptune runs.
 
         Args:
-            metrics: Dictionary with metric names as keys and measured quantities as values
-            step: Step number at which the metrics should be recorded, currently ignored
+            metrics: Dictionary with metric names as keys and measured quantities as values.
+            step: Step number at which the metrics should be recorded, currently ignored.
         """
-        assert rank_zero_only.rank == 0, 'experiment tried to log from global_rank != 0'
+        assert rank_zero_only.rank == 0, 'run tried to log from global_rank != 0'
 
         metrics = self._add_prefix(metrics)
         for key, val in metrics.items():
             # `step` is ignored because Neptune expects strictly increasing step values which
             # Lighting does not always guarantee.
-            self.log_metric(key, val)
+            self.experiment[key].log(val)
```

Comment:
`self.run[key].log(val)`

Reply:
@Herudaio do we not want to log metric in a namespace (e.g. `logs`) instead of logging them at the root of the run?

Reply:
Also, @shnela cover with tests